sdk/core/Azure.Core/src/DynamicData/DynamicData.AllowList.cs

```diff
@@ -0,0 +1,217 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text.Json;
+using Azure.Core.Json;
+
+namespace Azure.Core.Dynamic
+{
+    public partial class DynamicData
+    {
+        internal class AllowList
+        {
+            public static void AssertAllowedType<T>(T value)
+            {
+                if (value == null)
+                {
+                    return;
+                }
+
+                Type type = value.GetType();
```

Comment:
type is not used in this routine. Given that, should IsAllovedType take value and extract the type itself.

Reply:
If we do that, we would make the call to GetType() every time we recurse through this method.  If it's a no-op, that's fine.

Reply:
Addressed.